docs/spec/distribution/transactions.md

```diff
@@ -1,399 +1,208 @@
-    SetDelegatorDistribution(d.DelDistr)
-    SetGlobal(d.Global)
+``` 
+func (g Global) UpdateTotalValAccum(height int64, totalBondedTokens Dec) 
+    blocks = height - g.TotalValAccumUpdateHeight
+    g.TotalValAccum += totalDelShares * blocks
+    g.TotalValAccumUpdateHeight = height
 ```
 
-#### Delegation's entitlement to ValidatorDistribution.ProposerPool
+### Update validator's accums
 
-Delegations (including validator's self-delegation) are still subject
-commission on the rewards gained from the proposer pool.  Global shares in this
-context is actually the validators total delegations shares.  The recipient's
-shares is taken as the effective delegation shares less the validator's
-commission. 
+The total amount of delegator accum must be updated in order to determine the
+amount of pool tokens which each delegator is entitled to, relative to the
+other delegators for that validator. The accum is always additive to
+the existing accum. This term is to be updated each time a
+withdrawal is made from a validator. 
 
+``` 
+func (vi ValidatorDistInfo) UpdateTotalDelAccum(height int64, totalDelShares Dec) 
+    blocks = height - vi.TotalDelAccumUpdateHeight
+    vi.TotalDelAccum += totalDelShares * blocks
+    vi.TotalDelAccumUpdateHeight = height
 ```
-type DelegationFromProposerPool struct {
-    DelegationShares         sdk.Dec
-    ValidatorCommission      sdk.Dec
-    ValidatorDelegatorShares sdk.Dec
-    ValDistr                 ValidatorDistribution
-    DelDistr                 DelegatorDistribution
-}
-
-func (d DelegationFromProposerPool) DistributorTokens() DecCoins
-    return d.ValDistr.ProposerPool
-
-func (d DelegationFromProposerPool) DistributorCumulativeTokens() DecCoins
-    return d.ValDistr.EverReceivedProposerReward
 
-func (d DelegationFromProposerPool) DistributorPrevReceivedTokens() DecCoins
-    return d.ValDistr.PrevReceivedProposerReward
+### Global pool to validator pool
 
-func (d DelegationFromProposerPool) DistributorShares() sdk.Dec
-    return d.ValidatorDelegatorShares
+Every time a validator or delegator executes a withdrawal or the validator is
+the proposer and receives new tokens, the relevant validator must move tokens
+from the passive global pool to their own pool. It is at this point that the
+commission is withdrawn
 
-func (d DelegationFromProposerPool) DistributorPrevShares() sdk.Dec
-    return d.ValDistr.PrevDelegatorShares
-
-func (d DelegationFromProposerPool) RecipientShares() sdk.Dec
-    return d.DelegationShares * (1 - d.ValidatorCommission)
-
-func (d DelegationFromProposerPool) RecipientPrevShares() sdk.Dec
-    return d.DelDistr.PrevShares
-
-func (d DelegationFromProposerPool) RecipientAdjustment() sdk.Dec
-    return d.DelDistr.AdjustmentProposer
+``` 
+func (vi ValidatorDistInfo) TakeAccum(g Global, height int64, totalBonded, vdTokens, commissionRate Dec) g Global
+    g.UpdateTotalValAccum(height, totalBondedShares)
+    g.UpdateValAccum(height, totalBondedShares)
```

Comment:
What is g.UpdateValAccum?  (it's not documented... is it duplicate?)

Reply:
yes this is a duplicate of UpdateTotalValAccum, removed